sql: more log messages for debugging index backfill

vivekmenezes opened the pull request:
I'm seeing an index backfill on a GCE cluster not terminating, so I'd like to add these log messages to aid in debugging.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/13648)
<!-- Reviewable:end -->

RaduBerinde commented:
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[test, line 1 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13648#-KdBlrSDGlT10tkZl8sK:-KdBlrSDGlT10tkZl8sL:b-bhxh5y) ([raw file](https://github.com/cockroachdb/cockroach/blob/eb5ee65bac8a8a9957af023771334b28387bd99a/test#L1)):*
> ```
> panic: test timed out after 2m0s
> ```

is this file added by accident?

---

*[pkg/sql/distsqlrun/indexbackfiller.go, line 235 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13648#-KdBlxh6c2_Ny8baRsql:-KdBlxh6c2_Ny8baRsqm:b-2yu8y6) ([raw file](https://github.com/cockroachdb/cockroach/blob/eb5ee65bac8a8a9957af023771334b28387bd99a/pkg/sql/distsqlrun/indexbackfiller.go#L235)):*
> ```Go
> 	sp := work
> 	for row := int64(0); sp.Key != nil; row += chunkSize {
> 		log.VEventf(ctx, 2, "index add (%d, %d) at row: %d, span: %s",
> ```

Isn't always posting an event per chunk too much?  (mentioning just in case there is some confusion: `VEventf` is different than `if V(2) { Infof }`. It is equivalent to `if V(2) { Infof } else { Eventf }` meaning it will always generate an event).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13648)*
<!-- Sent from Reviewable.io -->

vivekmenezes committed:
sql: more log messages for debugging index backfill

vivekmenezes commented:
Review status: 0 of 2 files reviewed at latest revision, 2 unresolved discussions.

---

*[pkg/sql/distsqlrun/indexbackfiller.go, line 235 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13648#-KdBlxh6c2_Ny8baRsql:-KdBtl7jf_6DoqYx0s_B:bux0mi4) ([raw file](https://github.com/cockroachdb/cockroach/blob/eb5ee65bac8a8a9957af023771334b28387bd99a/pkg/sql/distsqlrun/indexbackfiller.go#L235)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

Isn't always posting an event per chunk too much?  (mentioning just in case there is some confusion: `VEventf` is different than `if V(2) { Infof }`. It is equivalent to `if V(2) { Infof } else { Eventf }` meaning it will always generate an event).
</blockquote></details>

Reverted back to using Infof. I thought it also conditionally logged an event

---

*[test, line 1 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/13648#-KdBlrSDGlT10tkZl8sK:-KdBtw5aMDuCHZU_SCEv:b-hfpv0l) ([raw file](https://github.com/cockroachdb/cockroach/blob/eb5ee65bac8a8a9957af023771334b28387bd99a/test#L1)):*
<details><summary><i>Previously, RaduBerinde wrote…</i></summary><blockquote>

is this file added by accident?
</blockquote></details>

removed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13648)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 2 files reviewed at latest revision, all discussions resolved, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/13648#-:-KdBwMGjWV5ePKv_Ti41:bnfp4nl)*
<!-- Sent from Reviewable.io -->
